src/parsers/espresso/7.2/parser.js

```diff
@@ -0,0 +1,292 @@
+        case "bohr":
+            scalingFactor = coefficients.BOHR_TO_ANGSTROM;
+            _units = ATOMIC_COORD_UNITS.cartesian;
+            break;
+        case "angstrom":
+            scalingFactor = 1.0;
+            _units = ATOMIC_COORD_UNITS.cartesian;
+            break;
+        case "crystal":
+            scalingFactor = 1.0;
+            _units = ATOMIC_COORD_UNITS.crystal;
+            break;
+        case "crystal_sg":
+            throw new Error("crystal_sg not supported yet");
+        default:
+            throw new Error(`Units ${units} not supported`);
+    }
+    return { _units, scalingFactor };
+}
+
+/**
+ * @summary Read atomic positions from ATOMIC_POSITIONS card
+ * @param {String} text - cards data
+ * @returns {{elements: Object[], coordinates: Object[], constraints: Object[], units: String}}
+ */
+function getAtomicPositions(text) {
+    const atomicSpeciesMatches = Array.from(text.matchAll(regex.atomicSpecies));
+    // eslint-disable-next-line no-unused-vars
+    const atomicSpecies = atomicSpeciesMatches.map((match) => ({
+        element: match[1],
+        mass: parseFloat(match[2]),
+        potential: match[3],
+    }));
+    const atomicPositionsMatches = Array.from(text.matchAll(regex.atomicPositions));
+    const units = text.match(regex.atomicPositionsUnits)[1];
+    const { _units, scalingFactor } = convertFromEspressoUnits(units);
+
+    const elements = atomicPositionsMatches.map((match, index) => ({
+        id: index,
+        value: match[1],
+    }));
+    const coordinates = atomicPositionsMatches.map((match, index) => ({
+        id: index,
+        value: [
+            parseFloat(match[2]) * scalingFactor,
+            parseFloat(match[3]) * scalingFactor,
+            parseFloat(match[4]) * scalingFactor,
+        ],
+    }));
+    const constraints = atomicPositionsMatches
+        .filter((match) => match[5] && match[6] && match[7]) // Check if all three constraints exist
+        .map((match, index) => ({
+            id: index,
+            value: [match[5] === "1", match[6] === "1", match[7] === "1"],
+        }));
+
+    return { elements, coordinates, constraints, units: _units };
+}
+
+/**
+ * @summary Converts ibrav value to cell type according to Quantum Espresso docs
+ * https://www.quantum-espresso.org/Doc/INPUT_PW.html#ibrav
+ * @param {Number} ibrav - ibrav parameter
+ * @returns {String}
+ */
+function ibravToCellType(ibrav) {
+    const type = IBRAV_TO_LATTICE_TYPE_MAP[ibrav];
+    if (type === undefined) {
+        throw new Error(`Invalid ibrav value: ${ibrav}`);
+    }
+    return type;
+}
+
+/**
+ * @summary Calculates cell parameters from celldm(i) or A, B, C parameters depending on which are present. Specific to Quantum Espresso.
+ * @param {Number[]} [celldm] - celldm(i) parameters
+ * @param {Number} [a] - A parameter
+ * @param {Number} [b] - B parameter
+ * @param {Number} [c] - C parameter
+ * @returns {Number[]}
+ */
+function getCellConstants(celldm, a, b, c) {
```

Comment:
Why are `a`, `b`, and `c` function arguments if they are just passed through? I think you could just use `celldm` here and this would also get rid of all the if statements:
```javascript
function getCellConstants(celldm) {
    const a = celldm[1] * coefficients.BOHR_TO_ANGSTROM;
    const b = celldm[2] * celldm[1] * coefficients.BOHR_TO_ANGSTROM;
    const c = celldm[3] * celldm[1] * coefficients.BOHR_TO_ANGSTROM;
    return [a, b, c];
```

Reply:
In QE A, B and C parameters can be supplied instead of celldm list. So, it's possible that this function receives only a,b,c and no celldm.